badge_server/main.py

```diff
@@ -76,518 +58,196 @@ def _get_pair_status_for_packages(pkg_sets):
-        commit_number=commit_number)
-    google_compat_res = _get_result_from_cache(
-        package_name=package_name,
-        badge_type=badge_utils.BadgeType.GOOGLE_COMP_BADGE,
-        commit_number=commit_number)
-    dependency_res = _get_result_from_cache(
-        package_name=package_name,
-        badge_type=badge_utils.BadgeType.DEP_BADGE,
-        commit_number=commit_number)
-
-    if self_compat_res['py3']['status'] == 'SUCCESS' and \
-        google_compat_res['py3']['status'] == 'SUCCESS' and \
-            dependency_res['status'] == 'UP_TO_DATE':
-        status = 'SUCCESS'
-    elif 'CALCULATING' in (
-        self_compat_res['py3']['status'],
-        google_compat_res['py3']['status'],
-        dependency_res['status']
-    ):
-        status = 'CALCULATING'
-    elif 'UNKNOWN' in (
-        self_compat_res['py3']['status'],
-        google_compat_res['py3']['status'],
-        dependency_res['status']
-    ):
-        status = 'UNKNOWN'
-    else:
-        status = 'CHECK_WARNING'
-
-    # Get the latest timestamp
-    self_ts = self_compat_res.get('timestamp', '')
-    google_ts = google_compat_res.get('timestamp', '')
-    dep_ts = dependency_res.get('timestamp', '')
-    ts_list = [self_ts, google_ts, dep_ts]
+    result_dict = badge_utils._build_default_result(
+        badge_utils.BadgeType.DEP_BADGE, 'UP_TO_DATE', {})
+
+    outdated_deps = badge_utils.highlighter.check_package(pkgname)
+    _deps_list = badge_utils.finder.get_deprecated_dep(pkgname)[1]
+    deprecated_deps = ', '.join(_deps_list)
+
+    details = {}
+    max_level = badge_utils.priority_level.UP_TO_DATE
+    for dep in outdated_deps:
+        dep_detail = {}
+        level = dep.priority.level
+        if level.value > max_level.value:
+            max_level = level
+        dep_detail['installed_version'] = dep.installed_version
+        dep_detail['latest_version'] = dep.latest_version
+        dep_detail['priority'] = dep.priority.level.name
+        dep_detail['detail'] = dep.priority.details
+        details[dep.name] = dep_detail
+        result_dict['status'] = max_level.name
+        result_dict['details'] = details
+        result_dict['deprecated_deps'] = deprecated_deps
+
+    result_dict['timestamp'] = datetime.datetime.now().strftime(
+        badge_utils.TIMESTAMP_FORMAT)
+
+    return result_dict
+
+
+def _get_status(
+        self_compat_res: dict,
+        google_compat_res: dict,
+        dependency_res: dict) -> str:
+    """Get the cummulative status"""
+    dep_status = dependency_res['status']
+    dep_status = 'SUCCESS' if dep_status == 'UP_TO_DATE' else dep_status
+
+    _statuses = []
+    for pyver in ['py2', 'py3']:
+        _statuses.append(self_compat_res[pyver]['status'])
+        _statuses.append(google_compat_res[pyver]['status'])
+    _statuses.append(dep_status)
+
+    if ['SUCCESS'] * len(_statuses) == _statuses:
+        return 'SUCCESS'
+    elif 'CALCULATING' in _statuses:
```

Comment:
Are all of these states still possible?

Reply:
I guess not - removed 'CALCULATING' and 'UNKNOWN'.